manage-data/data-store/data-streams/failure-store-recipes.md

```diff
@@ -0,0 +1,1153 @@
+
+To demonstrate this, we will follow a failed document through an unfamiliar data stream and ingest pipeline:
+```console
+POST my-datastream-ingest/_doc
+{
+    "@timestamp": "2025-04-21T00:00:00Z",
+    "important": {
+      "info": "The rain in Spain falls mainly on the plain"
+    }
+}
+```
+
+```console-result
+{
+  "_index": ".fs-my-datastream-ingest-2025.05.09-000001",
+  "_id": "F3S3s5YBwrYNjPmayMr9",
+  "_version": 1,
+  "result": "created",
+  "_shards": {
+    "total": 1,
+    "successful": 1,
+    "failed": 0
+  },
+  "_seq_no": 2,
+  "_primary_term": 1,
+  "failure_store": "used" // The document was sent to the failure store
+}
+```
+
+Now we search the failure store to check the failure document to see what went wrong.
+```console
+GET my-datastream-ingest::failures/_search
+```
+
+```console-result
+{
+  "took": 0,
+  "timed_out": false,
+  "_shards": {
+    "total": 1,
+    "successful": 1,
+    "skipped": 0,
+    "failed": 0
+  },
+  "hits": {
+    "total": {
+      "value": 1,
+      "relation": "eq"
+    },
+    "max_score": 1,
+    "hits": [
+      {
+        "_index": ".fs-my-datastream-ingest-2025.05.09-000001",
+        "_id": "F3S3s5YBwrYNjPmayMr9",
+        "_score": 1,
+        "_source": {
+          "@timestamp": "2025-05-09T06:24:48.381Z",
+          "document": {
+            "index": "my-datastream-ingest",
+            "source": { // When an ingest pipeline fails, the document stored is what was originally sent to the cluster.
+              "important": {
+                "info": "The rain in Spain falls mainly on the plain" // The important info that we failed to find was originally present on the document.
+              },
+              "@timestamp": "2025-04-21T00:00:00Z"
+            }
+          },
+          "error": {
+            "type": "illegal_argument_exception",
+            "message": "field [info] not present as part of path [important.info]", // The info field was not present when the failure occurred.
+            "stack_trace": """j.l.IllegalArgumentException: field [info] not present as part of path [important.info]
+	at o.e.i.IngestDocument.getFieldValue(IngestDocument.java:202)
+	at o.e.i.c.SetProcessor.execute(SetProcessor.java:86)
+	... 19 more
+""",
+            "pipeline_trace": [ // The first pipeline called the second pipeline.
+              "ingest-step-1",
+              "ingest-step-2"
+            ],
+            "pipeline": "ingest-step-2", // The document failed in the second pipeline.
+            "processor_type": "set" // It failed in the pipeline's set processor.
+          }
+        }
+      }
+    ]
+  }
+}
+```
+
+Despite not knowing the pipelines beforehand, we have some places to start looking. The `ingest-step-2` pipeline cannot find the `important.info` field despite it being present on the document that was sent to the cluster. If we pull that pipeline definition we find the following:
```

Comment:
```suggestion
Despite not knowing the pipelines beforehand, we have some places to start looking. The `ingest-step-2` pipeline cannot find the `important.info` field despite it being present in the document that was sent to the cluster. If we pull that pipeline definition we find the following:
```
Global comment: I think 'in' makes more sense than 'on' when referring to the contents of a document.